compiler-core/src/analyse.rs

```diff
@@ -1330,3 +1330,64 @@ fn get_compatible_record_fields<A>(
 
     compatible
 }
+
+/// Given a type, return a list of all the types it depends on
+fn get_type_dependencies(typ: &TypeAst) -> Vec<SmolStr> {
+    let mut deps = Vec::with_capacity(1);
+
+    match typ {
+        TypeAst::Var { .. } => (),
+        TypeAst::Hole { .. } => (),
+        TypeAst::Constructor {
+            name, arguments, ..
+        } => {
```

Comment:
We'll also need to take the module into account here. `Box`, `one.Box`, and `two.Box` are all different types.

Reply:
Good point, just added this and a test for it